examples/petsc/dmswarm.c

```diff
@@ -0,0 +1,990 @@
+// Copyright (c) 2017-2022, Lawrence Livermore National Security, LLC and other CEED contributors.
+// All Rights Reserved. See the top-level LICENSE and NOTICE files for details.
+//
+// SPDX-License-Identifier: BSD-2-Clause
+//
+// This file is part of CEED:  http://github.com/ceed
+
+//                        libCEED + PETSc DMSwarm Example
+//
+// This example demonstrates a simple usage of libCEED with DMSwarm.
+// This example combines elements of PETSc src/impls/dm/swam/tutorials/ex1.c and src/impls/dm/swarm/tests/ex6.c
+//
+// Build with:
+//
+//     make dmswarm [PETSC_DIR=</path/to/petsc>] [CEED_DIR=</path/to/libceed>]
+//
+// Sample runs:
+//
+//     ./dmswarm
+///
+//TESTARGS -ceed {ceed_resource} -test -dm_plex_dim 3 -dm_plex_box_faces 3,3,3 -dm_plex_box_lower -1.0,-1.0,-1.0 -dm_plex_simplex 0 -dm_plex_hash_location true -num_comp 2 -uniform_swarm
```

Comment:
Can we add either a second test or just an example invocation for the lumped mass case? And maybe label this one as KSP?

Reply:
Added in the lastest - good idea